libp2p/errors.nim

```diff
@@ -9,6 +9,9 @@ type
   # Base exception type for libp2p
   LPError* = object of CatchableError
 
+  ValidationResult* {.pure.} = enum
+    Accept, Reject, Ignore
+
```

Comment:
If we can avoid having this here, this is really gossipsub centric, and gossipsub shouldn't "escape" the pubsub folder
I know, cyclic imports are a pain :/

Reply:
Moved `ValidationResult` to a new file `libp2p/protocols/pubsub/errors.nim`.